### PR TITLE
improvement: update build system to make safer builds

### PR DESCRIPTION
**Pull request by maxime1992**
I've been using this library for quite a long time now :).

But there's one thing which is not really pleasant: Nearly every upgrade of Angular, I end up with a broken build, with this kind of message:
```
ERROR in Error: Error encountered resolving symbol values statically. Function calls are not supported. Consider replacing the function or lambda with a reference to an exported function (position 194:50 in the original .ts file), resolving symbol NgModule in /home/maxime/Documents/angular-starter/node_modules/angular2-prettyjson/node_modules/@angular/core/core.d.ts, resolving symbol PrettyJsonModule in /home/maxime/Documents/angular-starter/node_modules/angular2-prettyjson/src/prettyjson.module.ts, resolving symbol PrettyJsonModule in /home/maxime/Documents/angular-starter/node_modules/angular2-prettyjson/src/prettyjson.module.ts
    at positionalError (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler/bundles/compiler.umd.js:25266:35)
    at simplifyInContext (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler/bundles/compiler.umd.js:25109:27)
    at StaticReflector.simplify (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler/bundles/compiler.umd.js:25123:13)
    at StaticReflector.annotations (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler/bundles/compiler.umd.js:24553:41)
    at _getNgModuleMetadata (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_impl.js:138:31)
    at _extractLazyRoutesFromStaticModule (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_impl.js:109:26)
    at /home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_impl.js:129:27
    at Array.reduce (native)
    at _extractLazyRoutesFromStaticModule (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_impl.js:128:10)
    at /home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_impl.js:129:27
    at Array.reduce (native)
    at _extractLazyRoutesFromStaticModule (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_impl.js:128:10)
    at Object.listLazyRoutesOfModule (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_impl.js:53:22)
    at Function.NgTools_InternalApi_NG_2.listLazyRoutes (/home/maxime/Documents/angular-starter/node_modules/@angular/compiler-cli/src/ngtools_api.js:91:39)
    at AotPlugin._getLazyRoutesFromNgtools (/home/maxime/Documents/angular-starter/node_modules/@ngtools/webpack/src/plugin.js:207:44)
    at _donePromise.Promise.resolve.then.then.then.then.then (/home/maxime/Documents/angular-starter/node_modules/@ngtools/webpack/src/plugin.js:443:24)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)
```

I thought this was related only to angular being a dependency of the project but after giving a try it turns out that it was more than that.

Instead of removing the lib from my project I decided to get involved into this update of the build system :) I've been using [generator-angular2-library](https://github.com/jvandemo/generator-angular2-library) for a small [library](https://github.com/maxime1992/ngx-hover-opacity) I wanted to make and I was really happy with this starter.

So here it is, a proposal to have a better tooling for this lib :)

**TODO:**
- [  ] Make the tests work

### CODE REVIEW

**Comment by matiboy** [comment]
Hi @maxime1992 Thanks a lot for trying to fix the library. Any reason why the pending WIP? Looks like you've already made quite a bit of progress

**Comment by maxime1992** [comment]
Hi @matiboy, 

yes I don't know why but the tests are not running for now. I think I'm missing a karma config file or something like that.

Except for that, everything is working fine and I've been trying it (thanks to `yarn link`) on 2 different projects updated to angular 5 rc0 and it's working well :smile:

File: package.json [hunk 1]
```diff
@@ -20,32 +28,51 @@
...
+    "jasmine-core": "~2.5.2",
+    "jasmine-spec-reporter": "~3.2.0",
+    "karma": "~1.4.1",
+    "karma-chrome-launcher": "~2.0.0",
+    "karma-cli": "~1.0.1",
+    "karma-coverage-istanbul-reporter": "^0.2.0",
+    "karma-jasmine": "~1.1.0",
+    "karma-jasmine-html-reporter": "^0.2.2",
+    "mocha": "^3.5.1",
+    "node-sass": "^4.5.2",
```
Comment (matiboy) [inline package.json:63]: Hi @maxime1992 Any reason to include dependencies like node sass?

**Comment by matiboy** [comment]
@maxime1992 Thanks again for looking into this.
I created a branch called `update-build-system-for-safer-builds` which contains your changes plus a few commits from my side
For me, once I make those changes, the tests actually pass using `mocha spec`

**Comment by maxime1992** [comment]
@matiboy good to know! I'll definitely look into your branch but if tests are passing, feel free to merge in master :D! I've already tested it using `yarn link` within my projects and it was working great so if everything is green on your side... :tada: ;)

**Comment by maxime1992** [comment]
@matiboy Angular v5.0.0 should be out within less than a week. Could you merge? :smile:

**Review by matiboy — COMMENTED** [review]
@maxime1992 As mentioned, I had to make some changes which I put into another branch; did you get a chance to look at them and see whether it works for you with the changes?

File: package.json (hunk 1 above)
Comment (maxime1992) [reply package.json:63]: Probably not. I'll try to find some time this WE so we can finish this :).

**Comment by maxime1992** [comment]
So I've just looked at your commits and LGTM.
Do you want me to rebase my commit to remove the `[WIP DON'T MERGE]` or do you do it in your own branch? :)

**Comment by pfeigl** [comment]
Are you guys still looking into getting this done? :) Would be greatly appreciated!

**Comment by matiboy** [comment]
@pfeigl Actually yes, but using something a bit different, see [this branch](https://github.com/matiboy/angular2-prettyjson/tree/new-beginnings) based on https://github.com/robisim74/angular-library-starter

**Comment by pfeigl** [comment]
Very nice to hear. Don't want to make things more complicated, but can only recommand https://github.com/dherges/ng-packagr

It's actually incredible easy to build angular libraries with it - we are using it on quite alot of library projects we are doing.

**Comment by matiboy** [comment]
@pfeigl Can I bother you with a review of https://github.com/matiboy/angular2-prettyjson/pull/27
and/or installing the next version `npm install angular2-prettyjson@next` (should be 3.0.0) to confirm it's not causing problems anymore ?

**Comment by matiboy** [comment]
@maxime1992 Thank you so much for this PR; there was a lull after we discussed the tests problem and I've in the end moved to a completely new build based on Angular Library starter.
So I'll be closing this PR; thank you for your contribution nonetheless, appreciate it.

**Pull request closed without merging**